### ISSUE TITLE
Gerbers

### ISSUE THREAD

**Issue by qwertymodo** [defparam/21FX#1]
Can you post gerber files for those of us wanting to build our own?  Altium is a huge hassle just to set up the free trial long enough to export gerbers and then immediately uninstall.

**Comment by defparam** [comment]
I understand your pain. I have committed gerbers/drills for 21FX and snes-exp-bridge. Sorry for not doing that in advance. The aim of all my work is to keep everything completely open. If it is any constellation all my future designs will be done in KiCad (no more Altium unless absolutely necessary).

Thanks!
Evan

**Comment by qwertymodo** [comment]
Awesome, thanks. I've been wanting to build one for awhile, but Altium proved to be a barrier, not only due to it being non-free, but also the last time I tried, I wasn't able to figure it out before my trial expired.

**Comment by defparam** [comment]
Yeah, I have a couple of actions here I've been delaying for a while. (due to personal life and regular work)
1) do a toploader inline version of 21FX (probably named 21FX-TopLoader or 21FX-TL) with access and control to a lot more memory map and cart pins.
2) do a refresh of the original 21FX design porting it all to KiCad (to knock down the altium barrier) and fix potential cost/EOL issues with the CPLD (perhaps move to MAX V or a Xilinx variant)

In any case, good luck!
-Evan

**Comment by qwertymodo** [comment]
I actually built my own variant based on a Xilinx XC9572XL with a line driver on the data lines to handle the current requirements (as well as the 5V level shifting), but I didn't manage to get it working, and it was built in such a compact manner as to be incredibly difficult to debug. I'm hoping to build another one using the DB25 adapter that will be more debugger-friendly, and also I'm saving up to upgrade my oscilloscope to a MSO model with 16-channel logic analyser, which will help immensely. If I get it working, I'll send it your way (and if I don't, I'll probably bug you for help :P ).

**Comment by defparam** [comment]
Cool! my recommendation is definitely to attach an 8 or 16 channel logic analyzer to DATA[7:0] and PA[7:0]. The bus speed is reeaaally slow (3-4 MHz) so you can probably use a very cheap chinese one and interface it using Sigrok to observe the signals out of reset and see if the CPLD is performing the glitch when you expect it. I wouldn't necessarily trust a blind port of the HDL to Xilinx without having that visibility. Your currently prototype may still be able to work :)

**Comment by qwertymodo** [comment]
The 21FX gerbers are giving me errors trying to upload to OSHPark, it says the GKO (board outline) file is empty, so their upload processor can't find the board outline and refuses the upload.  The exp-bridge and snes-tap gerbers work just fine, it's just the 21FX that has issues.

**Comment by defparam** [comment]
OK sorry about that, I will test when I get home (currently at work). However if you are simply re-uploading my gerbers without modification I just opened up those projects on OSHPark for share:

21FX Rev C: https://oshpark.com/shared_projects/7yV9efbf
Exp Board: https://oshpark.com/shared_projects/r3Cxllq7

Those are the OSHPark orders I used for byuu's boards so they should be OK for order. When I get home I will fix the gerbers.

Thanks,
Evan
